pkg/kclient/generator/generators.go

```diff
@@ -0,0 +1,388 @@
+package generator
+
+import (
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	devfileParser "github.com/openshift/odo/pkg/devfile/parser"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+)
+
+const (
+	// DevfileSourceVolumeMount is the directory to mount the volume in the container
+	DevfileSourceVolumeMount = "/projects"
+
+	// EnvProjectsRoot is the env defined for project mount in a component container when component's mountSources=true
+	EnvProjectsRoot = "PROJECTS_ROOT"
+
+	// EnvProjectsSrc is the env defined for path to the project source in a component container
+	EnvProjectsSrc = "PROJECT_SOURCE"
+
+	deploymentKind       = "Deployment"
+	deploymentAPIVersion = "apps/v1"
+)
+
+// GetObjectMeta creates a common object meta
+func GetObjectMeta(name, namespace string, labels, annotations map[string]string) metav1.ObjectMeta {
+
+	objectMeta := metav1.ObjectMeta{
+		Name:        name,
+		Namespace:   namespace,
+		Labels:      labels,
+		Annotations: annotations,
+	}
+
+	return objectMeta
+}
+
+// ContainerParams is a struct that contains the required data to create a container object
+type ContainerParams struct {
+	Name         string
+	Image        string
+	IsPrivileged bool
+	Command      []string
+	Args         []string
+	EnvVars      []corev1.EnvVar
+	ResourceReqs corev1.ResourceRequirements
+	Ports        []corev1.ContainerPort
+}
+
+// getContainer creates a container spec that can be used when creating a pod
+func getContainer(containerParams ContainerParams) *corev1.Container {
```

Comment:
I think this should be a utils function

Reply:
True, I also moved `getServiceSpec()` since its the same like `getContainer()`